frappe/commands/site.py

```diff
@@ -70,29 +72,39 @@ def new_site(
 	"Create a new site"
 	from frappe.installer import _new_site
 
-	frappe.init(site=site, new_site=True)
+	rollback_callback = CallbackManager()
 
-	_new_site(
-		db_name,
-		site,
-		db_root_username=db_root_username,
-		db_root_password=db_root_password,
-		admin_password=admin_password,
-		verbose=verbose,
-		install_apps=install_app,
-		source_sql=source_sql,
-		force=force,
-		no_mariadb_socket=no_mariadb_socket,
-		db_password=db_password,
-		db_type=db_type,
-		db_host=db_host,
-		db_port=db_port,
-		db_user=db_user,
-		setup_db=setup_db,
-	)
+	try:
+		frappe.init(site=site, new_site=True)
+
+		_new_site(
+			db_name,
+			site,
+			db_root_username=db_root_username,
+			db_root_password=db_root_password,
+			admin_password=admin_password,
+			verbose=verbose,
+			install_apps=install_app,
+			source_sql=source_sql,
+			force=force,
+			no_mariadb_socket=no_mariadb_socket,
+			db_password=db_password,
+			db_type=db_type,
+			db_host=db_host,
+			db_port=db_port,
+			db_user=db_user,
+			setup_db=setup_db,
+			rollback_callback=rollback_callback,
+		)
+
+		if set_default:
+			use(site)
 
-	if set_default:
-		use(site)
+	except Exception:
+		traceback.print_exc()
+		if sys.__stdin__.isatty() and click.confirm("Do you want to rollback the failed site?", abort=True):
```

Comment:
We should clarify that site creation has failed in this message.

Reply:
I hope FC isn't faking a TTY, else it will just stay stuck forever on prompt :smile: